### DISCUSSION TITLE
How to programmatically navigate using React Router?

### DISCUSSION THREAD

**Discussion by Dark-Js1** [community/community#132529] [Programming Help] [answered]
With react-router I can use the Link element to create links which are natively handled by react router.

I see internally it calls this.context.transitionTo(...).

I want to do a navigation. Not from a link, but from a dropdown selection (as an example). How can I do this in code? What is this.context?

I saw the Navigation mixin, but can I do this without mixins?

**Comment by ghost** [comment]
Hi, use the "useHistory" hook. This.context is used to access the router context.

**Comment by goldieowner17** [comment]
hey @Dark-Js1 ! This is more suited for the Programming Help category as opposed to Discover. I've re-categorized this for you already.

**Comment by sanula12** [comment]
use the useNavigate hook

**Comment by nikolozkoiava** [comment, marked as answer]
The useHistory() hook is now deprecated. If you are using React Router 6, the proper way to navigate programmatically is as follows:

import { useHistory } from "react-router-dom";

function HomeButton() {
  const history = useHistory();

  function handleClick() {
    history.push("/home");
  }

  return (
    <button type="button" onClick={handleClick}>
      Go home
    </button>
  );
}

**Reply by Dark-Js1** [reply]
Thanks
